StreamChatClient_v3/Controllers/ChannelController.swift

```diff
@@ -3,14 +3,56 @@
 // Copyright © 2020 Stream.io Inc. All rights reserved.
 //
 
+import CoreData
 import Foundation
 
 /// `ChannelController` allows observing and mutating the controlled channel.
 ///
 ///  ... you can do this and that
 ///
-public class ChannelController<ExtraData: ExtraDataTypes>: Controller {
+public class ChannelController<ExtraData: ExtraDataTypes>: Controller, NSFetchedResultsControllerDelegate {
     // MARK: - Public
+    
+    public let channelId: ChannelId
+    public weak var delegate: ChannelControllerDelegate?
+    
+    public func startUpdating() {
+        fetchResultsController.delegate = self
+        try! fetchResultsController.performFetch()
```

Comment:
This is safe but, of course, it's nicer to do it the way you suggest. However, this will be a part of CIS-204, no need to worry about the controller at this point.